chore(docs): update contribution docs with service naming (#6486)
DataDog/dd-trace-py: docs/contributing-integrations.rst

The Pin API in ``ddtrace.pin`` is used to configure the instrumentation at runtime, including
enabling and disabling the instrumentation and overriding the service name.

Integration Checklists
------------------------
The lists here are necessary, but not sufficient to add an integration. Please read the full document.

General (all integrations):
[ ] Includes tests (see "Testing" in this document)
[ ] The CI suite is updated to include tests for the integration

Service Naming/Peer.Service (all integrations):
[ ] The integration has (a) a default service name and (b) supports service name schema

> [ ] The integration includes tests for the various service name schema (`default`, `V0 schema`, `V1`) which are sent to the test agent

Reviewer: ```suggestion
[ ] The integration includes tests for the various service name schema (`default`, `V0`, `V1`) which are sent to the test agent
```
?

    - `using it to instrument a later-added feature <https://github.com/DataDog/dd-trace-py/blob/96dc6403e329da87fe40a1e912ce72f2b452d65c/ddtrace/contrib/flask/patch.py#L149-L151>`_

Service Naming/Peer.Service
---------------------------

> Service Naming and Peer.Service help with automated service detection and modelling

Reviewer: ```suggestion
Service Naming and Peer.Service help with automated service detection and modeling
```

by providing different default values for the service and operation name span
attributes.  These attributes are adjusted by using environment variables which toggle 
various schemas. Every integration is required to support the existing schemas.

The API to support service name, peer.service, and schematization can be found here: <https://github.com/DataDog/dd-trace-py/blob/1.x/ddtrace/internal/schema/__init__.py#L52-L64>

The important elements are:
1. Every integration needs a default service name, which is what the service name for spans will be when the integration is called.

> 1. The default service name should be wrapped with `schematize_service_name()`

Reviewer: ```suggestion
2. The default service name should be wrapped with `schematize_service_name()`
```

> 1. If the span being created is a supported OTel format:

Reviewer: ```suggestion
3. If the span being created is a supported OTel format:
```

>     a. Wrap your operation name with the appropriate `schematize_*_operation` call (or add a new one)

Reviewer: can we add an e.g. here, like this:
```suggestion
    a. Wrap your operation name with the appropriate `schematize_*_operation` call e.g.         `_dbapi_span_operation_name=schematize_database_operation("postgres.query", database_provider="postgresql")`,
    (or add a new one) 
```

>     a. If OTel specifies precursors for peer.service, ensure your span includes those as tags

Reviewer: ```suggestion
    b. If OTel specifies precursors for peer.service, ensure your span includes those as tags
```

    - `using it to instrument a later-added feature <https://github.com/DataDog/dd-trace-py/blob/96dc6403e329da87fe40a1e912ce72f2b452d65c/ddtrace/contrib/flask/patch.py#L149-L151>`_

Service Naming/Peer.Service
---------------------------
Service Naming and Peer.Service help with automated service detection and modeling
by providing different default values for the service and operation name span
attributes.  These attributes are adjusted by using environment variables which toggle 
various schema. Every integration is required to support the existing schema.

The API to support service name, peer.service, and schema can be found here: <https://github.com/DataDog/dd-trace-py/blob/1.x/ddtrace/internal/schema/__init__.py#L52-L64>

The important elements are:
1. Every integration needs a default service name, which is what the service name for spans will be when the integration is called.
2. The default service name should be wrapped with `schematize_service_name()`
3. If the span being created is a supported Open Telemetry format:

  1. Wrap your operation name with the appropriate `schematize_*_operation` call (or add a new one)

>   2. If Open Telemetry specifies precursors for peer.service, ensure your span includes those as tags

Reviewer: ```suggestion
  2. If OpenTelemetry specifies precursors for peer.service, ensure your span includes those as tags
```
